app/helpers/remote_execution_helper.rb

```diff
@@ -38,6 +38,31 @@ def template_invocation_status(task, parent_task)
     end
   end
 
+  def template_invocation_api_status(task, parent_task)
+    if task.nil?
+      if parent_task.result == 'cancelled'
+        'cancelled'
+      else
+        'N/A'
+      end
+    elsif task.state == 'running'
+      'running'
+    elsif task.state == 'planned'
+      'planned'
+    else
+      case task.result
+        when 'warning', 'error'
+          'error'
+        when 'cancelled'
+          'cancelled'
+        when 'success'
+          'success'
+        else
+          task.result
+      end
```

Comment:
This could be reduced to "if result is warning, return error, otherwise just return the result"